client/components/TestPlanReportStatusDialog/index.jsx

```diff
@@ -0,0 +1,303 @@
+
+    const auth = evaluateAuth(me ?? {});
+    const { isSignedIn, isAdmin } = auth;
+
+    const requiredReports = useMemo(
+        () => getRequiredReports(testPlanVersion?.phase),
+        [testPlanVersion]
+    );
+
+    const [matchedReports, unmatchedTestPlanReports, unmatchedRequiredReports] =
+        useMemo(() => {
+            const matched = [];
+            const unmatchedTestPlan = [...testPlanReports];
+            const unmatchedRequired = [...requiredReports];
+
+            for (let i = 0; i < requiredReports.length; i++) {
+                for (let j = 0; j < testPlanReports.length; j++) {
+                    if (
+                        requiredReports[i].at.name ===
+                            testPlanReports[j].at.name &&
+                        requiredReports[i].browser.name ===
+                            testPlanReports[j].browser.name
+                    ) {
+                        if (testPlanReports[j].status === 'DRAFT') {
+                            matched.push({
+                                ...testPlanReports[j],
+                                metrics: getMetrics(testPlanReports[j])
+                            });
+                        } else {
+                            matched.push(testPlanReports[j]);
+                        }
+
+                        unmatchedTestPlan.splice(
+                            unmatchedTestPlan.indexOf(testPlanReports[j]),
+                            1
+                        );
+                        unmatchedRequired.splice(
+                            unmatchedRequired.indexOf(requiredReports[i]),
+                            1
+                        );
+                        break;
+                    }
+                }
+            }
+            return [matched, unmatchedTestPlan, unmatchedRequired];
+        }, [testPlanReports, requiredReports]);
+
+    const renderTableRow = (testPlanReport, required = 'Yes') => {
+        return (
+            <tr
+                key={`${testPlanReport.at.name}-${testPlanReport.browser.name}`}
+            >
+                <td>{required}</td>
+                <td>{testPlanReport.at.name}</td>
+                <td>{testPlanReport.browser.name}</td>
+                <td>{renderReportStatus(testPlanReport)}</td>
+            </tr>
+        );
+    };
+
+    const getMostRecentTestPlanRun = draftTestPlanRuns => {
+        let allTestResults = draftTestPlanRuns.flatMap(run => run.testResults);
+
+        let mostRecentTestResult = allTestResults.reduce(
+            (latestResult, currentResult) => {
+                let latestDate = new Date(latestResult.completedAt);
+                let currentDate = new Date(currentResult.completedAt);
+                return currentDate > latestDate ? currentResult : latestResult;
+            },
+            { completedAt: new Date(0) }
+        );
+
+        return mostRecentTestResult;
+    };
+
+    const getFormattedDate = dateString => {
```

Comment:
Could we make use of the `utils/formatter#convertDateToString` function instead? You should be able to pass the same `MMM DD, YYYY` format to it.

Reply:
Good catch. Didn't see that one but very happy to get to use moment.js here.